### PR TITLE
Highlight the current song on the album detail view

### PR DESCRIPTION
**Pull request by rivaldi8**
Until now, when playing an album, there was no way to view on the album detail view which song was playing. With these changes, now the current playing song is highlighted with a light blue background, like in the queue list. This way it's easier for the user to find the current point in the album.

### CODE REVIEW

File: shared/data/src/commonMain/kotlin/com/simplecityapps/shuttle/model/Song.kt [hunk 1]
```diff
@@ -71,4 +71,19 @@ data class Song(
...
+        other as Song
+
+        if (id != other.id) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return id.hashCode()
+    }
```
Comment (timusus) [inline shared/data/src/commonMain/kotlin/com/simplecityapps/shuttle/model/Song.kt:88]: I don't think it's a good idea to override the default behaviour of equals/hashcode here, as a convenience for comparing Song id's.

1.) This means that `Song` no longer behaves like a regular `data class`, which could be confusing / lead to unexpected behaviour
2.) I'm not sure if we're relying on the existing equality contract in other places, which could be broken by this change
3.) It's easy to just compare `song.id == otherSong.id`

Comment (rivaldi8) [reply shared/data/src/commonMain/kotlin/com/simplecityapps/shuttle/model/Song.kt]: I was a bit overwhelmed when a wrote that, so I don't think I noticed it was a data class. Anyway, I didn't remember about the data classes having already a proper `equals` ;-)

Fixed in 3a82613.

File: androidApp/main/app/src/main/java/com/simplecityapps/shuttle/ui/screens/library/albums/detail/AlbumDetailPresenter.kt [hunk 2]
```diff
@@ -111,8 +109,8 @@ class AlbumDetailPresenter @AssistedInject constructor(
     }
 
     override fun onQueuePositionChanged(oldPosition: Int?, newPosition: Int?) {
-        getCurrentSong()?.let {
-            view?.onCurrentSongChanged(it)
+        getCurrentSong()?.let { currentSong ->
+            view?.setData(this@AlbumDetailPresenter.songs, currentSong)
```
Comment (rivaldi8) [inline androidApp/main/app/src/main/java/com/simplecityapps/shuttle/ui/screens/library/albums/detail/AlbumDetailPresenter.kt:112]: Out of curiosity. Why is the qualifier needed in `this@AlbumDetailPresenter.songs`? I've tried and it seems to work without it.

By the way, nice work! These changes make it much simpler. Thanks!

Comment (timusus) [reply androidApp/main/app/src/main/java/com/simplecityapps/shuttle/ui/screens/library/albums/detail/AlbumDetailPresenter.kt:112]: Oh, `this@` is not needed, that looks like a copy paste error :D

**Review by timusus — APPROVED** [review]

**Pull request merged**